js/filters.js

```diff
@@ -114,7 +120,11 @@ weechat.filter('DOMfilter', ['$filter', '$sce', function($filter, $sce) {
             if (node === undefined || node === null) return;
             node = node.firstChild;
             while (node) {
-                var nextNode = process(node);
+                var nextNode;
+                // do not recurse inside links if the filter would create a nested link
+                if (!(createsAnchor && node.tagName == 'A')) {
```

Comment:
I get that a lot. Due to [how loose equality works in JS](https://github.com/getify/You-Dont-Know-JS/blob/master/types%20&%20grammar/ch4.md#loose-equals-vs-strict-equals), I prefer it in places where I know it will work correctly. And for example, the null check a couple of lines earlier would IMHO be more readable with loose equals: `(node === undefined || node === null)` vs. `(node == null)`.

But yeah, I'll change that. I'm not here to force my opinions. :)